keras_cv/layers/feature_pyramid.py

```diff
@@ -0,0 +1,208 @@
+
+    The final output of each level  will also have a conv2D operation
+    (usually with kernel = 3 and strides = 1).
+
+    The inputs to the layer should be a dict with int keys should match the
+    pyramid_levels, e.g. for `pyramid_levels` = [2,3,4,5], the expected input dict should
+    be `{2:c2, 3:c3, 4:c4, 5:c5}`.
+
+    The output of the layer will have same structures as the inputs, a dict with int keys
+    and value for each of the level.
+
+    Args:
+        pyramid_levels: a sorted python int list that specifies all the values of level
+            `i` for which the feature Ci will be specified.
+        num_channels: an integer representing the number of channels for the FPN
+            operations. Defaults to 256.
+        lateral_layers: a python dict with int keys that matches to each of the pyramid
+            level. The values of the dict should be `keras.Layer`, which will be called
+            with feature inputs from backbone at each level. Default to None, and a
+            `keras.Conv2D` layer with kernel 1x1 will be created for each pyramid level.
+        output_layers: a python dict with int keys that matches to each of the pyramid
+            level. The values of the dict should be `keras.Layer`, which will be called
+            with feature inputs and merged result from upstream levels. Default to None,
+            and a `keras.Conv2D` layer with kernel 3x3 will be created for each pyramid
+            level.
+        top_down_op: optional upsampling op between each layer. Default to None, and
+            `keras.layers.UpSampling2D` with 2x will be used.
+        merge_op: optional merge op for lateral result and upstream result. Default to
+            None, and `keras.layers.Add` will be used.
+
+    Sample Usage:
+    ```python
+
+    inp = tf.keras.layers.Input((384, 384, 3))
+    backbone = tf.keras.applications.EfficientNetB0(input_tensor=inp, include_top=False)
+    layer_names = ['block2b_add', 'block3b_add', 'block5c_add', 'top_activation']
+
+    backbone_outputs = {}
+    for i, layer_name in enumerate(layer_names):
+        backbone_outputs[i+2] = backbone.get_layer(layer_name).output
+
+    # output_dict is a dict with 2, 3, 4, 5 as keys
+    output_dict = keras_cv.layers.FeaturePyramid([2,3,4,5])(backbone_outputs)
+    ```
+    """
+
+    def __init__(
+        self,
+        pyramid_levels,
+        num_channels=256,
+        lateral_layers=None,
+        output_layers=None,
+        top_down_op=None,
+        merge_op=None,
+        **kwargs,
+    ):
+        super(FeaturePyramid, self).__init__(**kwargs)
+        self.pyramid_levels = sorted(pyramid_levels)
+        self.num_channels = num_channels
+
+        # required for successful serialization
+        self.lateral_layers_passed = lateral_layers
+        self.output_layers_passed = output_layers
+        self.top_down_op_passed = top_down_op
+        self.merge_op_passed = merge_op
+
+        if not lateral_layers:
```

Comment:
lets refactor the creation of the defaults and these big if else statements into helper functions.

i.e.

```
self.lateral_layers = _parse_lateral_layers(lateral_layers)
```